src/h3/qpack/mod.rs

```diff
@@ -53,6 +53,9 @@ pub enum Error {
 
     /// The decoded QPACK header name or value is not valid.
     InvalidHeaderValue,
+
+    /// The decoded QPACK header block exceeded the size limit.
+    HeaderBlockTooLarge,
```

Comment:
Bikeshedding warning, I would call this `HeaderListTooLarge`.

A "decoded header block" is a header list, and "encoded header list" is a header block. So to me, the only way a header block can be too large is if it exceeds the size of a stream (which should be caught as a frame size error in HEADERS).

In practice the more likely case is that header blocks are small but get decoded to something much bigger that blows out the limit of SETTINGS_MAX_HEADER_LIST_SIZE.

Reply:
Yeah, good point. Fixed.